test/jdk/javax/net/ssl/DTLS/DTLSOverDatagram.java

```diff
@@ -581,58 +577,55 @@ public final void runTest(DTLSOverDatagram testCase) throws Exception {
         }
     }
 
-    record ServerCallable(DTLSOverDatagram testCase, DatagramSocket socket,
-                          InetSocketAddress clientSocketAddr) implements Callable<String> {
-
-        @Override
-        public String call() throws Exception {
-            try {
-                testCase.doServerSide(socket, clientSocketAddr);
-            } catch (Exception e) {
-                System.out.println("Exception in  ServerCallable.call():");
-                e.printStackTrace(System.out);
-                serverException = e;
-
-                if (testCase.isGoodJob()) {
-                    throw e;
-                } else {
-                    return "Well done, server!";
-                }
-            }
+    String runServer(DTLSOverDatagram testCase, DatagramSocket socket,
+                          InetSocketAddress clientSocketAddr) throws Exception {
+        try {
+            serverStarted.countDown();
+            testCase.doServerSide(socket, clientSocketAddr);
+        } catch (Exception e) {
+            System.out.println("Exception in  ServerCallable.call():");
+            e.printStackTrace(System.out);
+            serverException = e;
 
             if (testCase.isGoodJob()) {
-                return "Well done, server!";
+                throw e;
```

Comment:
Why is this flipped, earlier if isGoodJob() returns false then we throw exception.